src/core/utils.ts

```diff
@@ -187,6 +191,88 @@ export function hasVisibleLegendItems(options: Highcharts.Options) {
   });
 }
 
+export function shouldShowSecondaryLegend(options: Highcharts.Options) {
+  // Note: While native Highcharts supports multiple legends, this
+  // implementation simplifies to at most 2 legends:
+  // - One group for all axes with opposite=false (primary axes)
+  // - One group for all axes with opposite=true (secondary axes)
+  //
+  // Important: A secondary legend should only be shown when BOTH primary and secondary
+  // series exist. This prevents abuse of the secondary legend API to force right-hand
+  // alignment when there's no actual dual-legend layout.
+  //
+  // Why we target yAxis only:
+  // In typical chart usage, dual-axis layouts use the y-axis for different scales
+  // (e.g., temperature on left, humidity on right). The x-axis is shared across all series.
+  // While Highcharts technically supports opposite x-axes, this is rare in practice and
+  // not supported by our dual-legend implementation.
+  //
+  // Why we support at most 2 axes:
```

Comment:
nit: I'd be a bit more focused with this comment as it is not a design doc. We can say e.g.: "The current implementation supports at most 2 axes, as there is currently no UX solution to handle more than that, although it is technically possible with Highcharts."